earthly/docs/Earthfile

```diff
@@ -0,0 +1,175 @@
+VERSION 0.7
+FROM python:3.11-bullseye
+
+mkdocs-material:
+    # Derived from official mkdocs-material docker container.
+    # https://github.com/squidfunk/mkdocs-material/blob/master/Dockerfile
+
+    # What version is this
+    ARG tag=latest
+    ARG registry
+
+    # Installation tmp directory
+    WORKDIR /poetry
+
+    # Environment Variables
+    ENV POETRY_HOME=/poetry
+    ENV PATH=$POETRY_HOME/bin:$PATH
+    ENV PYTHONDONTWRITEBYTECODE=1
+
+    # Install extra debian packages we will need to support plugins.
+    RUN apt-get update && \
+        apt-get install -y --no-install-recommends \
+        graphviz
+
+    # Install Poetry to manage python.
+    RUN curl -sSL https://install.python-poetry.org | python3 -
+    RUN poetry config installer.max-workers 10
+
+    # Copy our dependencies.
+    COPY pyproject.toml .
+    COPY poetry.lock .
+    COPY start.sh .
+
+    # Install it all with poetry
+    RUN poetry install --only main --no-root
```

Comment:
The `--no-root` is a no-op here, because there's no root project to begin with. 

Same thing with `-only main` because there's only a single dependency group (`main`). I can see leaving it here for the sake of future compatibility (i.e. not pulling in _future_ dev dependencies).